src/internal/signalr/handler.go

```diff
@@ -445,12 +408,127 @@ func (h *observationsHandler) ensureChargepointProps(srv chargepoint.Service, pr
 	return srv
 }
 
-func (h *observationsHandler) getParametersService() (parameters.Service, error) {
-	for _, service := range h.thing.Services(parameters.Parameters) {
+type energyHandler struct {
+	cache                 cache.Cache
+	thing                 adapter.Thing
+	lock                  sync.Mutex
+	confSrv               *config.Service
+	energyObservationChan chan model.Observation
+}
+
+func newEnergyHandler(cache cache.Cache, thing adapter.Thing, confSrv *config.Service) *energyHandler {
+	return &energyHandler{
+		cache:   cache,
+		thing:   thing,
+		confSrv: confSrv,
+		lock:    sync.Mutex{},
```

Comment:
If you don't create a pointer to the mutex, there's no need to instantiate it that way. Go utilizes "zero" values and a zero value for a struct is simply that struct. In other words, when you create a new `&energyHandler{}`, you already have `sync.Mutex{}` inside.